dts/arm/st/h7/stm32h743.dtsi

```diff
@@ -15,10 +15,35 @@
 			};
 		};
 	};
-	/* system data RAM accessible over over AXI bus */
+
+	/* System data RAM accessible over AXI bus: AXI SRAM in D1 domain */
 	sram0: memory@24000000 {
-		compatible = "mmio-sram";
 		reg = <0x24000000 DT_SIZE_K(512)>;
+		compatible = "mmio-sram";
+	};
+
+	/* System data RAM accessible over AHB bus: SRAM1 in D2 domain */
+	sram1: memory@30000000 {
+		reg = <0x30000000 DT_SIZE_K(128)>;
+		compatible = "mmio-sram";
+	};
+
+	/* System data RAM accessible over AHB bus: SRAM2 in D2 domain */
+	sram2: memory@30020000 {
+		compatible = "mmio-sram";
+		reg = <0x30020000 DT_SIZE_K(128)>;
+	};
+
+	/* System data RAM accessible over AHB bus: SRAM3 in D2 domain */
+	sram3: memory@30040000 {
+		compatible = "mmio-sram";
+		reg = <0x30040000 DT_SIZE_K(32)>;
+	};
```

Comment:
```suggestion
	/* System data RAM accessible over AHB bus: SRAM3 in D2 domain */
	sram3: memory@30040000 {
		compatible = "mmio-sram";
		reg = <0x30040000 DT_SIZE_K(24)>;
	};
	
	/* System data RAM accessible over AHB bus with MPU disabled: SRAM3 in D2 domain */
	non_mpu_sram: memory@30046000 {
		compatible = "mmio-sram";
		reg = <0x30046000 DT_SIZE_K(8)>;
	};
```

Reply:
I don't think DT definitions need to be related with MPU settings. MPU for a certain memory region may or may not be configured depending on the application or driver needs.

Reply:
> I don't think DT definitions need to be related with MPU settings. MPU for a certain memory region may or may not be configured depending on the application or driver needs.

@gmarull I agree with it. My idea here is to define or not define non_mpu_sram and get this memory region configured or not. If this memory region does not exist in DT, then everything works like usual. In other case non_mpu_sram will be accordingly configured.

Reply:
> @gmarull I agree with it. My idea here is to define or not define non_mpu_sram and get this memory region configured or not. If this memory region does not exist in DT, then everything works like usual. In other case non_mpu_sram will be accordingly configured.

Applications that may require full SRAM3 (ie 32K) will not work as usual, as you're suggesting to split it.

Reply:
> > @gmarull I agree with it. My idea here is to define or not define non_mpu_sram and get this memory region configured or not. If this memory region does not exist in DT, then everything works like usual. In other case non_mpu_sram will be accordingly configured.
> 
> Applications that may require full SRAM3 (ie 32K) will not work as usual, as you're suggesting to split it. 

@gmarull Sure, if non_mpu_sram is enabled. But we can properly document it. Otherwise, your ideas on MPU disabled sram region?

Reply:
It's up to the application how MPU is configured, not SRAM descriptions in DT.

Reply:
I also don't think we should split the SRAM3 in the DT. If we want to reserve only 8K for the Ethernet driver such that the rest can be used by other drivers, we can adapt the MPU region sizes and enforce our section to be exactly 8K in the linker script. Then other drivers can fill up the rest of SRAM3 with their own sections. I thought it would be okay for now as is because there is currently no other use case for SRAM3.

Reply:
Okay. Thanks for your ideas.